### PR TITLE
data(population): grapher dataset

### PR DESCRIPTION
**Pull request by lucasrodes**
Add a step to generate Grapher dataset for our Population OMM.

- This dataset was already half-way there (it was already in Garden).
- Currently, in Grapher, we use the variables from the "Key Indicator" dataset. This is not ideal, as we may deprecate this dataset in the future.
- This new dataset focuses on population metrics.

### CODE REVIEW

File: etl/steps/data/grapher/demography/2023-03-30/population.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    #
+    # Save outputs.
+    #
+    # Create a new grapher dataset with the same metadata as the garden dataset.
+    ds_grapher = create_dataset(dest_dir, tables=[tb_garden], default_metadata=ds_garden.metadata)
+
+    #
+    # Checks.
+    #
+    grapher_checks(ds_garden)
```
Comment (pabloarosado) [inline etl/steps/data/grapher/demography/2023-03-30/population.py:44]: Wouldn't it make more sense to apply the checks to `ds_grapher`?

File: etl/steps/data/grapher/demography/2023-03-30/population.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    )
+    return table
+
+
+def _modify_variable_description_historical(variable_description: str) -> str:
+    new_description = []
+    for line in variable_description.split("\n"):
+        match = re.search(r"([\d\-BCE,\s]*):.*", line)
+        if match:
+            if match.group(1) != "2022-2100":
```
Comment (pabloarosado) [inline etl/steps/data/grapher/demography/2023-03-30/population.py:123]: Maybe (thinking ahead) you should use `YEAR_THRESHOLD` instead of 2022.

File: etl/steps/data/grapher/demography/2023-03-30/population.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    new_description = "\n".join(new_description)
+    return new_description
+
+
+def _modify_variable_description_projection(variable_description: str) -> str:
+    new_description = []
+    for line in variable_description.split("\n"):
+        match = re.search(r"([\d\-BCE,\s]*):.*", line)
+        if match:
+            if match.group(1) == "2022-2100":
```
Comment (pabloarosado) [inline etl/steps/data/grapher/demography/2023-03-30/population.py:136]: Maybe (thinking ahead) you should use `YEAR_THRESHOLD` instead of 2022.

File: etl/steps/data/garden/demography/2022-12-08/population/meta.yml [hunk 4]
```diff
@@ -5,10 +5,14 @@ dataset:
   description: |
     This dataset has been created using multiple sources:
 
-      - We use UN WPP 2022 for data prior to 1950.
-      - Before that, we combine HYDE and Gapminder.
+    - UN WPP (2022): Covers the period 1950-2100. Future projections are based on the Medium variant.
+    - Gapminder (v6): Mostly covers the period 1800-1949. In addition, it complements the dataset with population data for the "Vatican" until 2100 (it is missing in UN WPP's estimates).
+    - HYDE (v3.2): Mostly covers the period 10,000 BCE - 1799. In addition, it complements the dataset with population for
+    some countries outside of this year period. E.g. it provides data for "Svalbard and Jan Mayen" and "Netherlands Antilles" until 2017.
```
Comment (pabloarosado) [inline etl/steps/data/garden/demography/2022-12-08/population/meta.yml:11]: Netherland Antilles dissolved in 2010, so this paragraph may cause some confusion.

File: etl/steps/data/garden/demography/2022-12-08/population/meta.yml [hunk 5]
```diff
@@ -5,10 +5,14 @@ dataset:
...
+    - Gapminder (Systema Globalis): Covers the period 1555-2008. It complements the dataset with data for former countries and other year-country tuples not present in the other sources.
```
Comment (pabloarosado) [inline etl/steps/data/garden/demography/2022-12-08/population/meta.yml:12]: For clarity, I'd rephrase `It complements the dataset with data for former countries and other year-country tuples not present in the other sources.`-> `It complements the dataset with former countries and other data points not present in the other sources.`.

File: etl/steps/data/garden/demography/2022-12-08/population/meta.yml [hunk 6]
```diff
@@ -45,10 +49,10 @@ tables:
         description: |
           Population by country, available from 10,000 BCE to 2100 based on Gapminder data, HYDE, and UN Population Division (2022) estimates.
 
-          10,000 BCE - 1799: Historical estimates by HYDE (v3.2).
-          1800-1949: Historical estimates by Gapminder.
-          1950-2021: Population records by the United Nations - Population Division (2022).
-          2022-2100: Projections based on Medium variant by the United Nations - Population Division (2022).
+          - 10,000 BCE - 1799: Historical estimates by HYDE (v3.2). Includes some datapoints Gapminder (Systema Globalis).
```
Comment (pabloarosado) [inline etl/steps/data/garden/demography/2022-12-08/population/meta.yml:52]: Includes some datapoints **from** Gapminder?

File: etl/steps/data/garden/demography/2022-12-08/population/meta.yml [hunk 7]
```diff
@@ -59,13 +63,18 @@ tables:
         description: |
           Share of the world's population by country, available from 10,000 BCE to 2100 based on Gapminder data, HYDE, and UN Population Division (2022) estimates.
 
-          10,000 BCE - 1799: Historical estimates by HYDE (v3.2).
-          1800-1949: Historical estimates by Gapminder.
-          1950-2021: Population records by the United Nations - Population Division (2022).
-          2022-2100: Projections based on Medium variant by the United Nations - Population Division (2022).
+          - 10,000 BCE - 1799: Historical estimates by HYDE (v3.2). Includes some datapoints Gapminder (Systema Globalis).
```
Comment (pabloarosado) [inline etl/steps/data/garden/demography/2022-12-08/population/meta.yml:66]: Given that you are using many `-`, maybe it's better to use `+` or `·` or `•` as bullet points. Alternatively, you could use [HTML tags](https://owid.slack.com/archives/C02706JNFC6/p1680185368251889), but I think it's better to avoid that in ETL metadata.

File: etl/steps/data/garden/demography/2022-12-08/population/meta.yml [hunk 8]
```diff
@@ -59,13 +63,18 @@ tables:
...
+          - 1800-1949: Historical estimates by Gapminder. Includes some datapoints from HYDE (v3.2) and Gapminder (Systema Globalis).
+          - 1950-2021: Population records by the United Nations - Population Division (2022). Includes some datapoints from HYDE (v3.2), Gapminder (Systema Globalis) and Gapminder (v6).
+          - 2022-2100: Projections based on Medium variant by the United Nations - Population Division (2022).
         unit: "%"
         short_unit: "%"
         display:
           name: Share of world population
           includeInTable: true
         sources: *population-sources
+      source:
```
Comment (pabloarosado) [inline etl/steps/data/garden/demography/2022-12-08/population/meta.yml:76]: Isn't this column dropped? (In line 69 of the step)

**Review by pabloarosado — APPROVED** [review]
Looks good! I included some minor corrections and suggestions.

Maybe for the future: Instead of having an incomplete metadata file and adding new variables programmatically, it could be better to add the metadata of all variables in the yaml file, having the year as a placeholder. We currently don't have a way to use placeholders in yaml (although I think there's an issue somewhere), but a workaround would be to have a helper function that takes the yaml file and replaces strings (e.g. 2022 -> 2023).

File: etl/steps/data/garden/demography/2022-12-08/population/meta.yml (hunk 8 above)
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2022-12-08/population/meta.yml:76]: This column is dropped in the Grapher step, but it is still available in Garden.
